src/components/errorbars/compute_error.js

```diff
@@ -0,0 +1,53 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+
+/**
+ * Compute the positive and negative error bar magnitudes.
+ *
+ * N.B. This function does not clean the dataPt entries, non-numeric
+ * entries result in undefined *error*
+ *
+ * @param {numeric} dataPt data point from where to compute the error magnitue
+ * @param {number} index index of dataPt in its corresponding data array
+ * @param {object} opts error bar attributes
+ *
+ * @return {array of two numbers}
+ *      - error[0] : error magnitude in the negative direction
+ *      - error[1] : " " " " positive "
+ */
+module.exports = function computeError(dataPt, index, opts) {
```

Comment:
Looks good to me. Don't need to do it now, but it occurs to me we could boost performance by just evaluating `opts` once and returning a more minimal function to evaluate in the loop:
```
function computeError(opts) {
    if(type === 'data') {
        if(opts.symmetric || opts.arrayminus === undefined) {
            var array = opts.array
            return function(dataPt, index) {
                var val = +(array[index]);
                return [val, val];
            }
        }
        // etc...
    }
}
```